tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -297,6 +298,239 @@ StatusOr<bool> FuseConvAlpha(HloComputation* comp) {
+  if (pattern_level == 0) {
+    final_instr = instr;
+  }
+
+  HloInstruction *op, *operand;
+  for (HloInstruction* user : instr->users()) {
+    if (pattern_level == 0) {
+      // Add
+      if (Match(user, m::AddAnyOrder(&op, m::Op(), m::Op(&operand)))) {
+        graph_string.AppendOp("add", op->shape().element_type());
+        operands.push_back(operand);
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+      // Scale
+      if (Match(user, m::MultiplyAnyOrder(&op, m::Op(),
+                                          m::Broadcast(m::Op(&operand))))) {
+        graph_string.AppendOp("scale", op->shape().element_type());
+        operands.push_back(operand);
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+      // Inverse Scale
+      if (Match(user, m::Divide(&op, m::Op(), m::Broadcast(m::Op(&operand))))) {
+        graph_string.AppendOp("invscale", op->shape().element_type());
+        operands.push_back(operand);
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+      // ReLU
+      if (Match(user, m::MaximumAnyOrder(&op, m::Op(),
+                                         m::Broadcast(m::ConstantScalar(0))))) {
+        graph_string.AppendOp("relu", op->shape().element_type());
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+    }
+
+    if (pattern_level == 1) {
+      // Convert with clamp
+      if (Match(user,
+                m::Convert(&op,
+                           m::Clamp(m::Broadcast(m::ConstantScalar()), m::Op(),
+                                    m::Broadcast(m::ConstantScalar()))))) {
+        graph_string.ChangeDataType(op->shape().element_type());
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+    }
+
+    // If none of the matches was successful and the pattern level is below the
+    // maximum level, attempt to match at higher level.
+    if (pattern_level < max_pattern_level) {
+      return CaptureConvGraphRecursive(user, operands, graph_string,
+                                       visited_instrs, final_instr,
+                                       pattern_level + 1);
+    }
+  }
+
+  // The first entry in the serialized graph is the convolution. The size is
+  // greater than one if at least one match was succesful.
+  if (graph_string.Size() > 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+// Captures the graph of pointwise operations operating on the convolution.
+bool CaptureConvGraph(HloInstruction* instr,
```

Comment:
It's a bit unclear what's effectively an argument and return value. I would return the relevant values like `operands` and `GraphString` as a tuple or struct.